web/components/roomConfig/GenericWebhookConfig.tsx

```diff
@@ -45,36 +39,27 @@ const ConnectionConfiguration: FunctionComponent<{
             name: nameRef?.current?.value || existingConnection?.config.name,
             ...(transFnEnabled ? { transformationFunction: transFn } : undefined),
         });
-        if (!existingConnection) {
-            // Clear fields
-            nameRef.current.value = "";
-            setTransFn(EXAMPLE_SCRIPT);
-            setTransFnEnabled(false);
-        }
-    }, [onSave, nameRef, transFn, setTransFnEnabled, setTransFn, existingConnection, transFnEnabled]);
+    }, [onSave, nameRef, transFn, existingConnection, transFnEnabled]);
 
     const onRemoveClick = useCallback(() => {
         onRemove();
     }, [onRemove]);
 
     const canEdit = !existingConnection || (existingConnection?.canEdit ?? false);
     return <div>
-        { !existingConnection && <div className={style.inputField}>
-            <label>Friendly name</label>
+        <InputField visible={!existingConnection} label="Friendly name" noPadding={true}>
             <input ref={nameRef} disabled={!canEdit} placeholder="My webhook" type="text" value={existingConnection?.config.name} />
-        </div> }
+        </InputField>
 
-        {!!existingConnection && <div className={style.inputField}>
-            <label>URL</label>
+        <InputField visible={!!existingConnection} label="URL" noPadding={true}>
             <input disabled={true} placeholder="URL hidden" type="text" value={existingConnection?.secrets?.url || ""} />
-        </div>}
+        </InputField>
 
-        { serviceConfig.allowJsTransformationFunctions && <div className={style.inputField}>
-            <label className={style.nopad}>Enable Transformation JavaScript</label>
+        <InputField visible={serviceConfig.allowJsTransformationFunctions} label="Enable Transformation JavaScript" noPadding={true}>
             <input disabled={!canEdit} type="checkbox" checked={transFnEnabled} onChange={() => setTransFnEnabled(!transFnEnabled)} />
```

Comment:
The first parameter for a `useState` set callback is the current value.
```suggestion
            <input disabled={!canEdit} type="checkbox" checked={transFnEnabled} onChange={useCallback(() => setTransFnEnabled(v => !v), [])} />
```